src/triggers.js

```diff
@@ -337,7 +337,17 @@ export function getResponseObject(request, resolve, reject) {
       }
       response = {};
       if (request.triggerName === Types.beforeSave) {
-        response['object'] = request.object._getSaveJSON();
+        try {
+          response['object'] = request.object._getSaveJSON();
+        } catch (error) {
+          // https://github.com/parse-community/parse-server/issues/7192
+          const e = resolveError(error, {
+            code: Parse.Error.VALIDATION_ERROR,
+            message:
+              'Even though the beforeSave script executed successfully without throwing exception, the returned object could not be converted to valid JSON.',
```

Comment:
Too verbose. I don't think it's necessary to mention what succeeded, but just mention clearly what failed any why, like: `afterSave failed because object could not be converted to JSON.`

Reply:
Oh, I think I copied from the below 'Script failed' and make up 'Script success'. I'll change just here.